python/ray/serve/tests/test_application.py

```diff
@@ -352,6 +352,20 @@ def test_deploy_from_yaml(self, serve_instance):
                 Application.from_yaml(tmp).to_dict(), app1.to_dict()
             )
 
+    def test_convert_to_import_path(self, serve_instance):
+        f = decorated_func.options(name="f")
+        C = DecoratedClass.options(name="C")
+        app = Application([f, C])
+
+        with tempfile.TemporaryFile(mode="w+") as tmp:
+            app.to_yaml(tmp)
+            tmp.seek(0)
+            reconstructed_app = Application.from_yaml(tmp)
```

Comment:
`app = Application.from_yaml(app.to_yaml())`

we added the `str` option for a reason! :)